pyproject.toml

```diff
@@ -0,0 +1,25 @@
+[build-system]
+requires = ["setuptools>=45", "wheel", "setuptools_scm>=6.2"]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools_scm]
```

Comment:
The version string for non-tagged commits using this method references the next pvlib version instead of the previous.  For example this PR branch currently references 0.9.2:
- `master`: `pvlib-0.9.1+18.g3f397ed4`
- this PR: `pvlib-0.9.2.dev16+g2c4fbe3f`

The number of commits since the last tag is `.devN` instead of `+N`.  "dirty" builds are also a bit different: `pvlib-0.9.2.dev16+g2c4fbe3f.d20220721` vs `pvlib-0.9.1+18.g3f397ed4.dirty`.

Not sure I see either of these as a problem, just pointing it out.